### PR TITLE
feat: Collider and Rigidbody physics

### PR DESCRIPTION
**Pull request by totisama**
When the player is moving in any direction we are detecting if he is colliding with something and blocking this direction.

### CODE REVIEW

File: Source/DataModels/Components/ComponentMeshCollider.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	void GetMinMaxPoints(std::vector<float3>& startingPoints, std::vector<float3>& points) const;
+
+private:
+
+	bool isTrigger;
+};
+
+inline bool ComponentMeshCollider::GetIsTrigger() const
+{
+	return this->isTrigger;
```
Comment (PaablooCH) [inline Source/DataModels/Components/ComponentMeshCollider.h:32]: `this` not necessary

File: Source/DataModels/Components/ComponentMeshCollider.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#include "ComponentMeshCollider.h"
+#include "ComponentTransform.h"
```
Comment (PaablooCH) [inline Source/DataModels/Components/ComponentMeshCollider.cpp:2]: componentTransform class is not used

File: Source/DataModels/Components/ComponentMeshCollider.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#include "GameObject/GameObject.h"
+#include "Application.h"
+
+#include "FileSystem/Json.h"
+
+#include "Geometry/LineSegment.h"
+#include "Geometry/Ray.h"
+#include "Physics/Physics.h"
+
+#include <iostream>
```
Comment (PaablooCH) [inline Source/DataModels/Components/ComponentMeshCollider.cpp:13]: why this library is used?

**Review by PaablooCH — CHANGES_REQUESTED** [review]
I see 3 things: 
PS: pls fill the description box

File: Source/DataModels/Components/ComponentMeshCollider.cpp (hunk 3 above)
Comment (totisama) [reply Source/DataModels/Components/ComponentMeshCollider.cpp]: Thanks, I didnt notice I was including it

**Review by OriolSors — DISMISSED** [review]
https://user-images.githubusercontent.com/31882514/232315751-8b4b0a08-18cd-4e18-ab09-baee212059aa.mp4

It seems that there is something weird with certain movements, maybe the raycasts are not accurate enough.

**Review by PaablooCH — APPROVED** [review]

**Comment by Dedado3000** [comment]
> axolotl-engine-2023-04-16-15-39-33_BVt18HOP.mp4 
> It seems that there is something weird with certain movements, maybe the raycasts are not accurate enough.

I solved in the last commit it was bc we weren't updating the transfoms matrices when moving

**Review by PaablooCH — COMMENTED** [review]
is this normal? xd
rampa

**Pull request merged**